docs/_includes/Types.md

```diff
@@ -1511,21 +1511,131 @@ where _members_ can include fields, functions, methods and declarations of neste
+have any cycles.
+
+A class or trait inherits (as if they are copied) all the instance members
+of its transitive parents. However, since names may not be overloaded in
+Dafny, different members (that is, members with different type signatures)
+within the set of transitive parents and the class or trait itself must have different names.[^overload]
+This restriction does mean that traits from different sources that
+coincidentally use the same name for different purposes cannot be combined
+by being part of the set of transitive parents for some new trait or class.
+
+Static members of a trait may not be redeclared and must be declared with a body.
+
+[^overload]: It is possible to conceive of a mechanism for disambiguating
+conflicting names, but this would add complexity to the language that does not
+appear to be needed, at least as yet.
+
+Where traits within an extension hierarchy do declare instance members with the same
+name (and thus the same signature), some rules apply. Recall that, for methods,
+every declaration includes a specification; if no specification is given
+explicitly, a default specification applies. Instance method declarations in traits,
+however, need not have a body.
+
+For a given non-static method M,
+
+* A trait or class may not redeclare M if it has a transitive parent that declares M and provides a body.
+* A trait may but need not provide a body if all its transitive parents that declare M do not declare a body.
+* A trait or class may not have more than one transitive parent that declares M with a body.
+* A class that has one or more transitive parents that declare M without a body
+and no transitive parent that declares M with a body must itself redeclare M
+with a body.
+* Currently (and under debate), the following restriction applies:
+Consider the DAG of classes and traits ordered by the _extends_ relationship
+and ending in a given class or trait CT. For a declaration of member M in CT,
+consider each path in the DAG that begins with a class or trait other than
+CT and that declares M, ends with CT, and does not pass through any trait
+that also declares M.
+The set of that begin these paths are the
+_direct parent declarors_ of M in CT.
+The restriction is that if this set has two or more members, then there must
+be one element of the set for which all of the other elements are
+its transitive parents.
+
+The last restriction above is the current implementation. It effectively limits
+inheritance of a method M to a single "chain" of declarations and does not
+permit mixins.
+
+Each of any method declarations explicitly or implicitly
+includes a specification. In simple cases, those syntactially separate
+specifications will be copies of each other (up to renaming to take account
+of differing formal parameter names). However they need not be. The rule is
+that the specifications of M in a given class or trait must be _no weaker than_
```

Comment:
Alternative, "no weaker than" could be mentioned as "as strong as".

Reply:
Revised